### PR TITLE
refactor owner list to accept tree or blob results

### PR DESCRIPTION
**Pull request by coury-clark**
Refactors the owner list so that it can be reused from a different context, notably from a tree instead of a blob. This will be useful for the repo level drilldown page.

## Test plan
Ensure ownership panel is the same after refactor
image

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+0.81 kb) | 0.01% (+0.81 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/cclark/own/refactor-owner-list/compare-report.html;tab=live_object) for a full comparison between the commits 69534c1 and 40bddb2 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/blob/own/FileOwnershipPanel.tsx [hunk 1]
```diff
@@ -60,17 +61,91 @@ export const FileOwnershipPanel: React.FunctionComponent<
...
+            if (owner.__typename === 'Person' && owner.user?.username) {
+                return `@${owner.user.username}`
+            }
+        }
+    }
+    return 'johndoe'
+}
+
+interface OwnerListProps {
+    data?: OwnershipConnectionFields | undefined
```
Comment (cbart) [inline client/web/src/repo/blob/own/FileOwnershipPanel.tsx:142]: According to [the other handbook](https://www.typescriptlang.org/docs/handbook/2/objects.html#optional-properties) a `?` property is already potentnially `undefined`

```suggestion
    data?: OwnershipConnectionFields
```

**Review by cbart — APPROVED** [review]
I really like this refactor.

File: client/web/src/repo/blob/own/FileOwnershipPanel.tsx [hunk 2]
```diff
@@ -60,17 +61,91 @@ export const FileOwnershipPanel: React.FunctionComponent<
...
+}
+
+const OwnerList: React.FunctionComponent<OwnerListProps> = ({ data }) => {
+    if (data?.nodes && data.nodes.length > 0) {
```
Comment (sashaostrikov) [inline client/web/src/repo/blob/own/FileOwnershipPanel.tsx:146]: nit: simplification
```suggestion
    if (data?.nodes.length) {
```

File: client/web/src/repo/blob/own/grapqlQueries.ts [hunk 3]
```diff
@@ -89,6 +89,51 @@ export const FETCH_OWNERS = gql`
     }
 `
 
+export const FETCH_TREE_OWNERS = gql`
```
Comment (sashaostrikov) [inline client/web/src/repo/blob/own/grapqlQueries.ts:92]: hmm, cannot see this being used 🤔

**Review by sashaostrikov — APPROVED** [review]
LGTM! Left a couple of comments.

File: client/web/src/repo/blob/own/grapqlQueries.ts (hunk 3 above)
Comment (coury-clark) [reply client/web/src/repo/blob/own/grapqlQueries.ts:92]: it isn't - yet. I just defined it now so we autogenerate the types, it'll be used as soon as this merges and I put up the PR for repo drilldown

File: client/web/src/repo/blob/own/FileOwnershipPanel.tsx (hunk 1 above)
Comment (coury-clark) [reply client/web/src/repo/blob/own/FileOwnershipPanel.tsx]: for some reason this is giving a type error without the explicit undefined, I'll poke at it some more. This is where my typescript knowledge is stretched 😄

Comment (coury-clark) [reply client/web/src/repo/blob/own/FileOwnershipPanel.tsx]: nevermind! must be something wrong with my local editor!

**Pull request merged**